google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java

```diff
@@ -81,6 +93,19 @@ protected BigQueryRpcFactory getDefaultRpcFactory() {
   }
 
   @Override
+  public TransportOptions getDefaultTransportOptions() {
+    return getDefaultHttpTransportOptions();
+  }
+
+  public static HttpTransportOptions getDefaultHttpTransportOptions() {
+    return HttpTransportOptions.newBuilder().build();
+  }
+
+  public HttpTransportOptions getHttpTransportOptions() {
```

Comment:
Looks suspicious and redundant, as there is already the getTransportOptions method (having two methods doing essentially same thing is potentially confusing for a user).

Please consider overriding getTransportOptions() here and returning HttpTransportOptions from it directly (also, similar to the previous comment, it can be moved to the HttpServiceOptions base class):

@Override
public /*works as of java spec 8.4.5.*/ HttpTransportOptions getTransportOptions() {
    return (HttpTransportOptions) super.getTransportOptions();
}